File: java/src/java/com/d_project/qrcode/Mode.java
package com.d_project.qrcode;

/**
 * モード.
 * @author Kazuhiko Arase 
 */
public interface Mode {

	/**
	 * 数値モード
	 */
	public static final int MODE_NUMBER 		= 1 << 0;

	/**
	 * 英数字モード
	 */
	public static final int MODE_ALPHA_NUM 	= 1 << 1;

	/**
	 * 8ビットバイトモード
	 */
	public static final int MODE_8BIT_BYTE	= 1 << 2;

	/**
	 * 漢字モード
	 */
	public static final int MODE_KANJI 		= 1 << 3;

}

File: ts/src/ts/com/d_project/qrcode/Mode.ts
/**
   * Mode
   * @author Kazuhiko Arase
   */
enum Mode {

    /**
     * number
     */
    MODE_NUMBER = 1 << 0,

    /**
     * alphabet and number
     */
    MODE_ALPHA_NUM = 1 << 1,

    /**
     * 8bit byte
     */
    MODE_8BIT_BYTE = 1 << 2,

    /**
     * KANJI
     */
    MODE_KANJI = 1 << 3
}

export default Mode;